mycore-base/src/main/java/org/mycore/datamodel/metadata/MCRObjectService.java

```diff
@@ -315,6 +301,67 @@ private void setDate(MCRMetaISO8601Date date) {
         }
     }
 
+    /**
+     * This method removes the date of the specified type from
+     * the date list.
+     * 
+     * @param type
+     *            a type as string
+     */
+    public final void removeDate(String type) {
+        if (DATE_TYPE_CREATEDATE.equals(type) || DATE_TYPE_MODIFYDATE.equals(type)) {
+            LOGGER.error("Cannot delete built-in date: " + type);
+        } else {
+            MCRMetaISO8601Date d = getISO8601Date(type);
+            if (d != null) {
+                dates.remove(d);
+            }
+        }
+    }
+
+    /**
+     * This method sets the status classification
+     */
+    public final void setState(MCRCategoryID state) {
+        if (state == null) {
+            this.state = state;
+        } else {
+            if (MCRCategoryDAOFactory.getInstance().exist(state)) {
+                this.state = state;
+            } else {
+                LOGGER.error("Error at setting servstate classification.",
```

Comment:
We should 1. Throw an error and abort  or 2. use logger.warn. Not really related to this PR.